### DISCUSSION TITLE
xml files not being saved in LocalizationResources folder

### DISCUSSION THREAD

**Discussion by julianadormon** [LazZiya/XLocalizer#44] [Q&A]
@LazZiya Sorry I'm not sure if you saw my last comment on our last discussion as it was closed.
While the changes you helped mw with regarding switching the Culture worked, once I run the Console app and see that the translations are made successfully, no xml translation files are being written to the LocalizationResources folder. Any ideas? Thanks!

This is my set-up:

            using System.Globalization;
            using Microsoft.Extensions.DependencyInjection;
            using Microsoft.Extensions.Localization;
            using GqlClient;
            using IdentityModel.Client;
            using Microsoft.AspNetCore.Builder;
            using Microsoft.AspNetCore.Localization;
            using Microsoft.Extensions.Configuration;
            using XLocalizer.Translate;
            using XLocalizer.Translate.MyMemoryTranslate;
            using XLocalizer;
            using XLocalizer.Xml;
            using XLocalizer.Common;
            using NotificationDaemon.LocalizationResources;
            
            namespace NotificationDaemon
            {
                public class Program
                {
                    public static void Main(string[] args)
                    {
                        // Build and configure Services
                        var Services = new ServiceCollection();
                        
                        ConfigureServices(Services);
            
                        // Add additional Services specific to the console app if needed
                        //Services.AddSingleton<LocalizerService>(); // Add LocalizerService as a singleton
            
                        var ServiceProvider = Services.BuildServiceProvider();
            
                        // Use the service provider to resolve and run your application logic
            
                        var Localizer = ServiceProvider.GetRequiredService<IStringLocalizer<LocSource>>();
            
                        //Test different cultures below by uncommenting them
                        
                        using (CultureSwitcher cs = new("es"))
                        {
                            var LocalizedString = Localizer["Hello world"];
                            Console.WriteLine(LocalizedString);
                        }
            
                        using (CultureSwitcher cs = new("fr"))
                        {
                            var LocalizedString = Localizer["The color is blue."];
                            Console.WriteLine(LocalizedString);
                        }
            
                        using (CultureSwitcher cs = new("en-US"))
                        {
                            var LocalizedString = Localizer["Hello world"];
                            Console.WriteLine(LocalizedString);
                        }
                    }
            
                    private static void ConfigureServices(IServiceCollection Services)
                    {
                        // Add IConfiguration instance
                        var configuration = new ConfigurationBuilder()
                            .SetBasePath(AppDomain.CurrentDomain.BaseDirectory)
                            .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
                            .Build();
            
                        Services.AddSingleton<IConfiguration>(configuration);
                        
                        Services.AddHttpClient<ITranslator, MyMemoryTranslateService>();
                        Services.AddSingleton<IXResourceProvider, XmlResourceProvider>();
            
                        Services.AddRazorPages().AddXLocalizer<LocSource, MyMemoryTranslateService>(ops =>
                        {
                            ops.ResourcesPath = "LocalizationResources";
                            ops.AutoAddKeys = true;
                            ops.AutoTranslate = true;
                            ops.UseExpressMemoryCache = true;
                        });
                    }
                }
            }

And here is the Service class:

          using Microsoft.Extensions.Localization;
          using System.Globalization;
          
          namespace NotificationDaemon
          {
              public class LocalizerService
              {
                  private readonly IStringLocalizerFactory _localizerFactory;
          
                  public LocalizerService(IStringLocalizerFactory localizerFactory)
                  {
                      _localizerFactory = localizerFactory;
                  }
          
                  public string GetString(string key, CultureInfo culture)
                  {
                      var localizer = _localizerFactory.Create(typeof(LocalizationResources.LocSource));
                      return localizer[key, culture];
                  }
              }
          }

**Comment by julianadormon** [comment]
@LazZiya I tried to sponsor this project but it says it no longer is taking PayPal. Anyway you can send me your PayPal email address to julian[at]proofpix.com ?

**Reply by LazZiya** [reply]
Thank you for your kindness :) PayPal is not working in my country :( 
There is GitHub sponsors option, not sure if it will work 😅

**Reply by julianadormon** [reply]
Unfortunately, I am in Panama and I cannot use credit cards on GitHub. My card keeps getting declined. My only options is to use PayPal. So sorry!

**Reply by LazZiya** [reply]
No problem, at least you are the first to offer a sponsorship, so consider it done :)

**Comment by LazZiya** [comment]
https://github.com/LazZiya/XLocalizer/discussions/43#discussioncomment-8181150
